src/internal_events/sematext_metrics.rs

```diff
@@ -1,46 +1,62 @@
+use super::prelude::{error_stage, error_type};
 use metrics::counter;
 use vector_core::internal_event::InternalEvent;
 
 use crate::event::metric::Metric;
 
 #[derive(Debug)]
-pub struct SematextMetricsInvalidMetricReceived<'a> {
+pub struct SematextMetricsInvalidMetricError<'a> {
     pub metric: &'a Metric,
 }
 
-impl<'a> InternalEvent for SematextMetricsInvalidMetricReceived<'a> {
+impl<'a> InternalEvent for SematextMetricsInvalidMetricError<'a> {
     fn emit_logs(&self) {
-        warn!(
+        error!(
             message = "Invalid metric received; dropping event.",
+            error_code = "invalid_metric",
+            error_type =  error_type::ENCODER_FAILED,
+            stage = error_stage::PROCESSING,
             value = ?self.metric.value(),
             kind = ?self.metric.kind(),
-            internal_log_rate_secs = 30,
+            internal_log_rate_secs = 10,
         )
     }
 
     fn emit_metrics(&self) {
         counter!(
             "processing_errors_total", 1,
-            "error_type" => "invalid_metric",
+            "error_code" => "invalid_metric",
+            "error_type" => error_type::ENCODER_FAILED,
+            "stage" => error_stage::PROCESSING,
         );
     }
 }
 
 #[derive(Debug)]
-pub struct SematextMetricsEncodeEventFailed {
-    pub error: &'static str,
+pub struct SematextMetricsEncodeEventError<E> {
+    pub error: E,
 }
 
-impl InternalEvent for SematextMetricsEncodeEventFailed {
+impl<E: std::fmt::Display> InternalEvent for SematextMetricsEncodeEventError<E> {
     fn emit_logs(&self) {
-        warn!(
-             message = "Failed to encode event; dropping event.",
-             error = %self.error,
-             internal_log_rate_secs = 30
+        error!(
+            message = "Failed to encode event; dropping event.",
+            error = %self.error,
+            error_code = "sematext_metrics_encoding_failed",
```

Comment:
Same here, could we make this error code more generalized?

Reply:
This will be dropped considering it doesn't bring any information 😉